### PR TITLE
Timer feature initial UI

### PR DESCRIPTION
**Pull request by mehedihasan88**
Timer Feature Developed( TimerFragment.java + fragment_timer.xml)
fragment_home.xml is updated simply ( List elements of project is bit rounded now with some padding and and title is all capital with some top margin)

### CODE REVIEW

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        super.onViewCreated(view, savedInstanceState);
+        chronometer= view.findViewById(R.id.chronometer);
+        toggleButton=view.findViewById(R.id.toggleButton);
+        toggleButton.setChecked(true);
+        toggleButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean isRunning) {
+                System.out.println(isRunning);
+                if(isRunning){
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: remove this line

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import com.example.teamtracker.R;
+import com.example.teamtracker.database.RoomDB;
+import com.example.teamtracker.models.Project;
+import com.example.teamtracker.models.Task;
+
+public class TimerFragment extends Fragment {
+
+    Chronometer chronometer;
+    ToggleButton toggleButton;
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: make this params private

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            public void onCheckedChanged(CompoundButton compoundButton, boolean isRunning) {
+                System.out.println(isRunning);
+                if(isRunning){
+
+                    chronometer.stop();
+                    long duration = SystemClock.elapsedRealtime() - chronometer.getBase();
+                    System.out.println(duration);
+                    showAddTaskDialog(getContext(),duration);
+                    chronometer.setBase(SystemClock.elapsedRealtime());
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: remove this line

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                }else {
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: remove this line

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                    chronometer.setBase(SystemClock.elapsedRealtime());
+                    chronometer.start();
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: remove this line

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                }
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java:76]: remove this extra line

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+            }
+        });
+    }
+
+    private void showAddTaskDialog(Context context,long duration) {
+        final EditText projectNameEditText = new EditText(context);
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: rename this to taskNameEditText

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+                    chronometer.setBase(SystemClock.elapsedRealtime());
+                    chronometer.start();
+
+                }
+
+            }
+        });
+    }
+
+    private void showAddTaskDialog(Context context,long duration) {
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: Context context,long duration

give a space after ',' 
also use Long object instead primitive long

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        duration%=(60*60); // remaining seconds
+        long minute = duration/60; // calculating minute
+        long sec =  duration%60; //calculating second
+        // making the time string
+        if(hour != 0) time+= hour + "h ";
+        if(minute != 0) time+= minute + "m ";
+        if(sec != 0) time+= sec + "s";
+        else if(hour == 0 && sec == 0) time+= sec + "s";
+        return time;
+    }
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: give proper spaces to them.
like 
duration = duration / 1000;
duration = duration % (60*60)

you have to also remove this 1000, 60*60 from here. these are called here magic number. instead you will take const variable like SECONDS_IN_ONE_HOUR = 60*60 and use SECONDS_IN_ONE_HOUR here

File: TeamTracker/app/src/main/res/drawable/circle_shape.xml [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" >
+    <stroke
+        android:width="1dp"
+        android:color="@color/colorPrimaryDark" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/circle_shape.xml]: remove extra line

File: TeamTracker/app/src/main/res/drawable/circle_shape.xml [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    <solid android:color="#ffffff" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/circle_shape.xml]: remove extra line

File: TeamTracker/app/src/main/res/drawable/circle_shape.xml [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    <padding
+        android:left="1dp"
+        android:right="1dp"
+        android:bottom="1dp"
+        android:top="1dp" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/circle_shape.xml]: remove extra line

File: TeamTracker/app/src/main/res/drawable/rounded_corner_menu_colored_rectangle.xml [hunk 13]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" >
+    <stroke
+        android:width="1dp"
+        android:color="@color/colorPrimary" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/rounded_corner_menu_colored_rectangle.xml]: remove this extra line

File: TeamTracker/app/src/main/res/drawable/rounded_corner_menu_colored_rectangle.xml [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+    <solid android:color="@color/colorPrimary" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/rounded_corner_menu_colored_rectangle.xml]: remove this extra line

File: TeamTracker/app/src/main/res/drawable/rounded_corner_menu_colored_rectangle.xml [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+    <padding
+        android:left="1dp"
+        android:right="1dp"
+        android:bottom="1dp"
+        android:top="1dp" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/rounded_corner_menu_colored_rectangle.xml]: remove this extra line

File: TeamTracker/app/src/main/res/drawable/rounded_corner_white_rectangle.xml [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" >
+    <stroke
+        android:width="1dp"
+        android:color="@color/colorWhite" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/rounded_corner_white_rectangle.xml]: remove this extra line

File: TeamTracker/app/src/main/res/drawable/rounded_corner_white_rectangle.xml [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+    <solid android:color="#ffffff" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/rounded_corner_white_rectangle.xml]: remove this extra line

File: TeamTracker/app/src/main/res/drawable/rounded_corner_white_rectangle.xml [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+    <padding
+        android:left="1dp"
+        android:right="1dp"
+        android:bottom="1dp"
+        android:top="1dp" />
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/drawable/rounded_corner_white_rectangle.xml]: remove this extra line

File: TeamTracker/app/src/main/res/layout/fragment_timer.xml [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+            android:background="@drawable/timer_toggle_button_selector"
+            android:checked="true"
+            android:textOff=""
+            android:textOn=""
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/chronometer" />
+
+
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/layout/fragment_timer.xml:53]: remove this extra line

File: TeamTracker/app/src/main/res/values/strings.xml [hunk 20]
```diff
@@ -11,4 +11,5 @@
     <string name="nav_home">Home</string>
     <string name="logout">Logout</string>
     <string name="nav_demo">Demo Menu</string>
+    <string name="nav_timer">Timer</string>
```
Comment (sajjad-h) [inline TeamTracker/app/src/main/res/values/strings.xml]: remove as no need

**Comment by mehedihasan88** [comment]
Hey I have resolved all the suggested changes @sajjad-h

**Review by sajjad-h — APPROVED** [review]

**Comment by sajjad-h** [comment]
Great Job @mehedihasan88 
@alif-cse-ju will also check this pull request and will merge this. Thank you.

**Review by alifalhasan — APPROVED** [review]
Good job @mehedihasan88

File: TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java (hunk 7 above)
Comment (mehedihasan88) [reply TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: I did not get it!

Comment (mehedihasan88) [reply TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java]: Oh i get it :|

**Pull request merged**